server/blob/blob-union/src/main/java/org/apache/james/blob/union/UnionBlobStore.java

```diff
@@ -196,6 +196,13 @@ boolean streamHasContent(PushbackInputStream pushBackIS) throws IOException {
         return Mono.empty();
     }
 
+    @Override
+    public Mono<Void> delete(BucketName bucketName, BlobId blobId) {
+        return Mono.defer(() -> currentBlobStore.delete(bucketName, blobId))
+            .and(legacyBlobStore.delete(bucketName, blobId))
+            .onErrorResume(this::logAndReturnEmpty);
```

Comment:
as I understand you used `.and()` which is greedy, the mono execute both deletion from current & legacy and combines the results. But using `logAndReturnEmpty()` doesn't seem to match with what the log says:
```
error happens from current blob store, fall back to legacy blob store
```

Reply:
Greedy in what sense? That what the same for `deleteBucket`, we want the blob to be deleted in both current AND legacy if it exists in both. I agree that we can adapt the log message though for this case